Lib/idlelib/sidebar.py

```diff
@@ -1,19 +1,33 @@
 """Line numbering implementation for IDLE as an extension.
 Includes BaseSideBar which can be extended for other sidebar based extensions
 """
+import contextlib
 import functools
 import itertools
 
 import tkinter as tk
+from tkinter.font import Font
 from idlelib.config import idleConf
 from idlelib.delegator import Delegator
 
 
+def get_lineno(text, index):
+    """Utility to get the line number of an index in a Tk text widget."""
+    return int(float(text.index(index)))
+
+
 def get_end_linenumber(text):
     """Utility to get the last line's number in a Tk text widget."""
-    return int(float(text.index('end-1c')))
+    return get_lineno(text, 'end-1c')
```

Comment:
The docstrings violate pep8 and only 1 function is needed.
```
def get_lineno(text, idex='end-1c'):
    "Return line number in text of idex, which default to the last line"
    return int(float(text.index(idex)))

Reply:
I guess it's a matter of style, but IMO having `get_lineno()` return the last line number by default isn't an intuitive default behavior. It feels like favoring conciseness over clarity, which I usually try to avoid.

Reply:
@terryjreedy:
> The docstrings violate pep8

As far as I can tell, this would not be in violation of [PEP-257's section on one line docstrings](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings) though. PEP-257 is main authority on docstrings. If anything, it indicates that triple quotes are preferable over single quotes:

> Triple quotes are used even though the string fits on one line. This makes it easy to later expand it.

Reply:
This module currently consistently uses triple-quotes even for single-line doc-strings.

I'd be happy to change that in a follow-up PR, but I think it's better to avoid making more widespread changes in this (already large) PR, and I wouldn't want to break the current consistency.

Reply:
Leave the triple quote if you prefer and space allows.  I am more concerned about an unneeded quarter-liner.